### PR TITLE
Automate test maintenance when updating components

### PR DESCRIPTION
**Pull request by JonZeolla**
# Contributor Comments

This PR runs `pipenv run invoke update` after adding some helper logic to maintain the tests when updating components.  We have a test file that specifies a terraform `required_version`, but it must be aligned with the default/built-in version of terraform from `easy_infra.yml`.

## Pull Request Checklist

Thank you for submitting a contribution to `easy_infra`!

In order to streamline the review of your contribution we ask that you review and comply with the below requirements:

- [x] Rebase your branch against the latest commit of the target branch
- [x] If you are adding a dependency, please explain how it was chosen
- [x] If manual testing is needed in order to validate the changes, provide a testing plan and the expected results
- [x] If there is an issue associated with your Pull Request, link the issue to the PR.

### CODE REVIEW

File: easy_infra/config.py [hunk 1]
```diff
@@ -45,29 +45,29 @@ def write_config(*, config: dict, config_file: Path):
         dump(config, file)
 
 
-def update_config_file(*, thing: str, version: str):
```
Comment (JonZeolla) [inline easy_infra/config.py:48]: This is just a rename for consistency based on the prior refactor

File: tasks.py [hunk 2]
```diff
@@ -23,7 +23,6 @@
 from easy_infra import __project_name__, __version__, config, constants, utils
 from tests import test as run_test
 
-# TODO: Does this work on intel?
```
Comment (JonZeolla) [inline tasks.py:26]: We verified that this does work on intel in #184

File: tasks.py [hunk 3]
```diff
@@ -457,23 +456,30 @@ def update(_c, debug=False):
...
     for repo in constants.GITHUB_REPOS_TAGS:
         version = utils.get_latest_tag_from_github(repo=repo)
-        config.update_config_file(thing=repo, version=version)
+        config.update_config_file(package=repo, version=version)
 
     for project in constants.HASHICORP_PROJECTS:
         version = utils.get_latest_release_from_hashicorp(project=project)
-        config.update_config_file(thing=project, version=version)
+        config.update_config_file(package=project, version=version)
+        if project == "terraform":
```
Comment (JonZeolla) [inline tasks.py:472]: Although `terraform` is the only member of `constants.HASHICORP_PROJECTS` right now, we don't want to assume that will always be the case.

File: tests/test.py [hunk 4]
```diff
@@ -708,7 +710,7 @@ def run_terraform(*, image: str) -> None:
             "terraform plan || false",
             1,
         ),  # Not supported; reproduce "Too many command line arguments. Configuration path expected." error
-        #     locally with `docker run -e DISABLE_SECURITY=true -v $(pwd)/tests/terraform/tool/checkov:/iac seiso/easy_infra:latest terraform plan
+        #     locally with `docker run -e DISABLE_SECURITY=true -v $(pwd)/tests/terraform/tool/checkov:/iac seiso/easy_infra:latest-terraform terraform plan
```
Comment (JonZeolla) [inline tests/test.py:713]: This was a miss from #184; we no longer have a `-latest` image.

**Review by digeseiso — APPROVED** [review]

**Pull request merged**